xbmc/epg/Epg.cpp

```diff
@@ -749,10 +749,11 @@ bool CEpg::IsRemovableTag(const CEpgInfoTag &tag) const
 bool CEpg::LoadFromClients(time_t start, time_t end)
 {
   bool bReturn(false);
-  CPVRChannelPtr channel = Channel();
-  if (channel)
+
+  CSingleLock lock(m_critSection);
```

Comment:
and this one actually slows things down when something else tries to access the epg while an update is running